src/main/java/Bob.java

```diff
@@ -0,0 +1,207 @@
+                    Task task = getTask(fromIndex, toIndex);
+                    list.add(task);
+                    say("added: " + task);
+                }
+            }
+
+            private static Task getTask(int fromIndex, int toIndex) {
+                String desc = fromIndex < toIndex
+                                ? argument.substring(0, fromIndex)
+                                : argument.substring(0, toIndex);
+                desc = desc.strip();
+                String from = fromIndex < toIndex
+                                ? argument.substring(fromIndex + 6, toIndex)
+                                : argument.substring(fromIndex + 6);
+                from = from.strip();
+                String to = fromIndex < toIndex
+                                ? argument.substring(toIndex + 4)
+                                : argument.substring(toIndex + 4, fromIndex);
+                to = to.strip();
+                if (desc.isBlank()) {
+                    throw new MissingArgumentException("description of the event");
+                }
+                return new Event(desc, from, to);
+            }
+        },
+        MARK("mark") {
+            @Override
+            public void run() {
+                if (argument.isBlank()) {
+                    throw new MissingArgumentException("index of the task that you want to mark");
+                }
+                int i;
+                try {
+                    i = Integer.parseInt(argument) - 1;
+                } catch (NumberFormatException e) {
+                    throw new IncorrectArgumentException("an integer");
+                }
+                if (i < 0 || i >= list.size()) {
+                    throw new IncorrectArgumentException("a valid index");
+                }
+                list.get(i).mark();
+                say("Nice! I've marked this task as done:\n  " +
+                        list.get(i));
+            }
+        },
+        UNMARK("unmark") {
+            @Override
+            public void run() {
+                if (argument.isBlank()) {
+                    throw new MissingArgumentException("index of the task that you want to unmark");
+                }
+                int i;
+                try {
+                    i = Integer.parseInt(argument) - 1;
+                } catch (NumberFormatException e) {
+                    throw new IncorrectArgumentException("an integer");
+                }
+                if (i < 0 || i >= list.size()) {
+                    throw new IncorrectArgumentException("a valid index");
+                }
+                list.get(i).unmark();
+                say("OK, I've marked this task as not done yet:\n  " +
+                        list.get(i));
+            }
+        },
+        CATCH_ALL("") {
+            @Override
+            public void run() {
+                throw new UnknownCommandException();
+            }
+        };
+
+        public final String CMD;
```

Comment:
I think the coding standards mentioned that we should avoid short forms? Maybe can name as COMMAND instead of CMD?